app/helpers/federation_helper.rb

```diff
@@ -124,4 +124,25 @@ def federation_external_class?(class_object)
   end
 
 
+  def canonical_portal(ontologies)
+    portals = federated_portals.map{|portal| portal.first}
+    portal_counts = Hash.new(0)
+    # Count occurrences of each portal in the pull_location URL
+      portals.each do |portal|
+        if ontology[:pullLocation]&.include?(portal.to_s)
```

Comment:
return an error because `ontology` is not defined.

Reply:
When you did the merge of development, the line of ontologies loop changed position in a weird way. Fixed now.